public/plugin.ts

```diff
@@ -12,14 +12,33 @@ import {
 } from '../../../src/core/public';
 import { PLUGIN_NAME, ROUTES } from './utils/constants';
 import { SecurityAnalyticsPluginSetup, SecurityAnalyticsPluginStart } from './index';
+import { Plugin as DataPublicPlugin } from '../../../src/plugins/data/public';
+
+export interface SecurityAnalyticsPluginSetupDeps {
+  data: DataPublicPlugin;
+}
+export interface SecurityAnalyticsPluginStartDeps {
+  data: DataPublicPlugin;
+}
 
 export class SecurityAnalyticsPlugin
-  implements Plugin<SecurityAnalyticsPluginSetup, SecurityAnalyticsPluginStart> {
+  implements
+    Plugin<
+      SecurityAnalyticsPluginSetup,
+      SecurityAnalyticsPluginStart,
+      SecurityAnalyticsPluginSetupDeps,
+      SecurityAnalyticsPluginStartDeps
+    > {
   constructor(private readonly initializerContext: PluginInitializerContext) {
     // can retrieve config from initializerContext
   }
 
-  public setup(core: CoreSetup): SecurityAnalyticsPluginSetup {
+  private plugins?: SecurityAnalyticsPluginStartDeps;
```

Comment:
We should call this `startupPlugins` to be clear that these will contain the start plugins. Thoughts?

Reply:
I found proper way:
<img width="481" alt="image" src="https://user-images.githubusercontent.com/1564068/213860347-94441bb8-faf7-49e0-9f14-1fdc380e924c.png">

[Here](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/1.2/src/core/CONVENTIONS.md#applications) is the reference.